### DISCUSSION TITLE
How the references work?

### DISCUSSION THREAD

**Discussion by horizon0514** [tldraw/tldraw-v1#566] [Q&A] [answered]
https://github.com/tldraw/tldraw/blob/main/packages/core/src/inputs.ts#L4
![image](https://user-images.githubusercontent.com/1518608/146121427-86421eb5-0640-4273-8893-a98f6881f87d.png)
I want to know how the Line4 import works. 
There is no `paths` setting in the tsconfig.

Does anyone familiar with the tsconfig can help me?

**Comment by steveruizok** [comment, marked as answer]
Hey, moving this over to discussion. You can read more about project references [here](https://www.typescriptlang.org/docs/handbook/project-references.html). 

The problem with using paths in a monorepo is that they need to be cleaned up at build time, or else the aliases path will be present in the distributed types (which will not work for consumers). For example, tsc will leave an import like `import { Something } from "~types"` exactly like that, meaning we need to do the work ourselves of turning it into a relative path, e.g. `import { Something } from "../../types"`. Using references allows us to script against .tsconfig without risking turning our monorepo imports into relative paths. We definitely _dont_ want `import { Vec } from "@tldraw/vec"` turning into `import { Vec } from "../../vec/src/index.ts"` or something.

**Reply by sdwlig** [reply]
This makes sense in the source, and for when webpack et al is being used.  However, for use without doing a webpack or similar build, relative paths need to be used.  While proposed, agreement on a browser-level path search equivalent hasn't happened yet.  Most projects use a build tool, but I have built a big complex project without it.
